src/MongrationDotNet/MigrationRunner.cs

```diff
@@ -29,33 +31,54 @@ public async Task Migrate()
                 logger?.LogInformation(LoggingEvents.MigrationStarted,
                     "Migration started for type: {type}, version: {version} and description: {description} ",
                     migrationCollection.Type, migrationCollection.Version.ToString(), migrationCollection.Description);
-                
-                var latestAppliedMigration = await migrationDetailsCollection
+
+            var latestAppliedMigration = await migrationDetailsCollection
                     .Find(x => x.Version == migrationCollection.Version)
                     .FirstOrDefaultAsync();
 
-                if (latestAppliedMigration != null)
-                {
-                    logger?.LogInformation(LoggingEvents.MigrationSkipped,
-                        "Migration has already been applied. Skipped migration for type: {type}, version: {version} and description: {description} ",
-                        migrationCollection.Type, migrationCollection.Version.ToString(),
-                        migrationCollection.Description);
-                    continue;
-                }
+            if (latestAppliedMigration != null)
+            {
+                logger?.LogInformation(LoggingEvents.MigrationSkipped,
+                    "Migration has already been applied or in progress. Skipped migration for type: {type}, version: {version} and description: {description} ",
+                    migrationCollection.Type, migrationCollection.Version.ToString(),
+                    migrationCollection.Description);
+                continue;
+            }
 
-                await SetMigrationInProgress(migrationCollection);
-                migrationCollection.Prepare();
-                await migrationCollection.ExecuteAsync(database, logger);
-                await SetMigrationAsCompleted(migrationCollection);
-                logger?.LogInformation(LoggingEvents.MigrationCompleted,
-                    "Migration completed type: {type}, version: {version} and description: {description} ",
-                    migrationCollection.Type, migrationCollection.Version.ToString(), migrationCollection.Description);
+            await SetMigrationInProgress(migrationCollection);
+            migrationCollection.Prepare();
+            await migrationCollection.ExecuteAsync(database, logger);
+            await SetMigrationAsCompleted(migrationCollection);
```

Comment:
Do a try catch around prepare & execute steps and if either throws then mark the migration as failed in the database.

Reply:
It is already done. Part of other PR